cmake/clang-tidy.cmake

```diff
@@ -0,0 +1,36 @@
+option(ENABLE_CLANG_TIDY "Add run-clang-tidy automatically to builds" OFF)
```

Comment:
Please rename the option to SPDLOG_ENABLE_TIDY

Reply:
OK